tests/tests_pytorch/checkpointing/test_model_checkpoint.py

```diff
@@ -1171,6 +1171,61 @@ def test_ckpt_version_after_rerun_same_trainer(tmpdir):
     assert set(os.listdir(tmpdir)) == expected
 
 
+def test_ckpt_version_counter_disabled_after_rerun_new_trainer(tmpdir):
+    """Check that previous checkpoints get overwritten and no suffixes are generated when new trainer instances are
+    used."""
+    epochs = 2
+    for i in range(epochs):
+        mc = ModelCheckpoint(
+            dirpath=tmpdir, save_top_k=-1, monitor="epoch", filename="{epoch}", enable_version_counter=False
+        )
+        trainer = Trainer(
+            max_epochs=epochs,
+            limit_train_batches=1,
+            limit_val_batches=1,
+            default_root_dir=tmpdir,
+            callbacks=[mc],
+            logger=False,
+            enable_progress_bar=False,
+            enable_model_summary=False,
+        )
+        trainer.fit(BoringModel())
+
+        # check best_k_models state
+        expected = {"epoch=0.ckpt", "epoch=1.ckpt"}
+        assert {Path(f).name for f in mc.best_k_models} == expected
+
+    # check created ckpts
+    actual = {f.basename for f in tmpdir.listdir()}
+    assert actual == {"epoch=0.ckpt", "epoch=1.ckpt"}
+
+
+def test_ckpt_version_counter_disabled_after_rerun_same_trainer(tmpdir):
```

Comment:
What value does this test add that isn't added by the one above?

Reply:
I thought that it might be useful to check with the same trainer. But after all, it looks redundant. Removing.